talend-component-maven-plugin/src/it/web/.jenkins/Jenkinsfile

```diff
@@ -247,49 +215,92 @@ pipeline {
+      when {
+        expression { runtimeVersion.contains('-SNAPSHOT') }
+      }
+      steps {
+        withCredentials([_NEXUS_CREDENTIALS]) {
+          script {
+            sh "bash ${_SCRIPTS_PATH}/tcomp-build-fast.sh ${_ROOT_PATH}/pom.xml"
+          }
+        }
+      }
+    }
+    stage('Prepare component-runtime server') {
+      steps {
+        withCredentials([_NEXUS_CREDENTIALS]) {
+          script {
+            sh """
+              bash "${_SCRIPTS_PATH}/server-registry-init.sh" "${_DOWNLOAD_DIR}" \
+                                                              "${_INSTALL_DIR}" \
+                                                              "${_COVERAGE_DIR}" \
+                                                              "${runtimeVersion}" \
+                                                              '1.41.0' \
+                                                              '/root/.m2/repository' \
+                                                              "${_SERVER_PORT}"
+              """.stripIndent()
+          }
+        }
+      }
+      post {
+        always {
+          println "Artifact SETENV: ${_ARTIFACT_SETENV}\n"
+          archiveArtifacts artifacts: "${_ARTIFACT_SETENV}", allowEmptyArchive: false, onlyIfSuccessful: false
+          println "Artifact REGISTRY: ${_ARTIFACT_REGISTRY}\n"
+          archiveArtifacts artifacts: "${_ARTIFACT_REGISTRY}", allowEmptyArchive: false, onlyIfSuccessful: false
+        }
+      }
+    }
+    stage('Start component-runtime server') {
       steps {
         withCredentials([_NEXUS_CREDENTIALS]) {
           script {
-            if (runtimeVersion.contains('-SNAPSHOT')) {
-              sh """ "${_SCRIPTS_PATH}/tcomp-build.sh" "${_ROOT_PATH}/pom.xml" """
-            }
-
-            // TODO: TCOMP-2253 move from "azure-dls-gen2" to local build of the demo connector
             sh """
-              "${_SCRIPTS_PATH}/server-registry-init.sh" "${_DOWNLOAD_DIR}" "${_INSTALL_DIR}" "${_COVERAGE_DIR}" \
-                                                         "${runtimeVersion}" \
-                                                         "1.36.0" "azure-dls-gen2" \
-                                                         "${_SERVER_PORT}"
-              "${_SCRIPTS_PATH}/server-registry-start.sh" "${_INSTALL_DIR}" "${_COVERAGE_DIR}" "${_SERVER_PORT}"
-            """
+              bash "${_SCRIPTS_PATH}/server-registry-start.sh" "${_INSTALL_DIR}" \
+                                                               "${_COVERAGE_DIR}" \
+                                                               "${_SERVER_PORT}"
+            """.stripIndent()
+          }
+        }
+      }
+    }
+    stage('Check component-runtime server') {
+      steps {
+        withCredentials([_NEXUS_CREDENTIALS]) {
+          script {
+            sh "bash ${_SCRIPTS_PATH}/server-check.sh ${_SERVER_PORT}"
           }
         }
       }
     }
     stage('Execute Test') {
```

Comment:
I split server check because it allows to see quickly where an issue is coming from.
If it is a server config issue / starting server issue / a build issue
It also allows in Jenkins to see the log separately and to act separately.
We will be able in the future to send a different warning to teams.